packages/compiler/core/program.ts

```diff
@@ -574,24 +573,31 @@ export async function compile(
     const basedir = getDirectoryPath(file.file.path);
     await loadImports(
       file.statements.filter(isImportStatement).map((x) => ({ path: x.path.value, target: x })),
-      basedir
+      basedir,
+      getSourceFileScope(file.file)
     );
   }
 
+  function getSourceFileScope(sourcefile: SourceFile): SourceFileScope {
+    return sourceFileScopes.get(sourcefile) ?? { type: "project" };
```

Comment:
Is this default needed, and is it the correct default? Could consider throwing, or else `"unknown"` may be a safer default?